main.go

```diff
@@ -246,3 +256,87 @@ func bindFlags(cmd *cobra.Command, v *viper.Viper) {
 		}
 	})
 }
+
+// Create an issue in the provider repo with a message describing the upgrade failure
+func createFailureIssue(ctx upgrade.Context, repoOrg string, repoName string) (string, error) {
+	now := time.Now()
+	hr, min, _ := now.Clock()
+	yr, mth, day := now.Date()
+	minStr := fmt.Sprintf("%v", min)
+	if min < 10 {
+		minStr = fmt.Sprintf("0%v", min)
+	}
+	title := fmt.Sprintf("Upgrade provider failure: %v %v, %v @ %v:%v", mth, day, yr, hr, minStr)
+
+	getLatestWorkflowRun := exec.CommandContext(ctx, "gh", "run", "list",
+		"--workflow="+"upgrade-provider.yml",
+		"--repo="+repoOrg+"/"+repoName,
+		"--limit=1",
+		"--json=url",
+	)
+	bytes1 := new(bytes.Buffer)
+	getLatestWorkflowRun.Stdout = bytes1
+	err := getLatestWorkflowRun.Run()
+	if err != nil {
+		return "createFailureIssue error: failed to query latest workflow run", err
+	}
+	workflowRunUrls := []struct {
+		Url string `json:"url"`
+	}{}
+	err = json.Unmarshal(bytes1.Bytes(), &workflowRunUrls)
+	if err != nil {
+		return "createFailureIssue error: failed to unmarshal `gh run list` output", err
+	}
+	if len(workflowRunUrls) == 0 {
+		return "createFailureIssue error: no workflow runs detected", err
+	}
+	workflowUrl := workflowRunUrls[0].Url
+
+	getIssues := exec.CommandContext(ctx, "gh", "search", "issues",
+		"Upgrade provider failure: ",
+		"--repo="+repoOrg+"/"+repoName,
+		"--json=title,number",
```

Comment:
We only want open issues here.
```suggestion
		"--json=title,number",
		"--state=open",
```